packages/arb-token-bridge-ui/src/components/TransferPanel/TokenSearch.tsx

```diff
@@ -255,43 +289,44 @@ function TokenRow({ style, onClick, token }: TokenRowProps): JSX.Element {
             <div className="flex flex-col items-start space-y-1">
               {/* TODO: anchor shouldn't be nested within a button */}
               {isDepositMode ? (
-                <a
-                  href={`${getExplorerUrl(l1Network.id)}/token/${
-                    token.address
-                  }`}
-                  target="_blank"
-                  rel="noopener noreferrer"
-                  className="text-xs text-blue-link underline"
-                  onClick={e => e.stopPropagation()}
-                >
-                  {shortenAddress(token.address).toLowerCase()}
-                </a>
+                <>
+                  {isL2NativeToken ? (
+                    <BlockExplorerTokenLink
+                      chain={l2Network}
+                      address={token.address}
+                    />
+                  ) : (
+                    <BlockExplorerTokenLink
+                      chain={l1Network}
+                      address={token.address}
+                    />
+                  )}
+                </>
               ) : (
                 <>
                   {tokenHasL2Address ? (
-                    <a
-                      href={`${getExplorerUrl(l2Network.id)}/token/${
-                        token.l2Address
-                      }`}
-                      target="_blank"
-                      rel="noopener noreferrer"
-                      className="text-xs text-blue-link underline"
-                      onClick={e => e.stopPropagation()}
-                    >
-                      {token.l2Address
-                        ? shortenAddress(token.l2Address).toLowerCase()
-                        : ''}
-                    </a>
+                    <BlockExplorerTokenLink
+                      chain={l2Network}
+                      address={token.l2Address}
+                    />
                   ) : (
                     <span className="text-xs text-gray-900">
-                      This token hasn&apos;t been bridged to L2
+                      This token hasn&apos;t been bridged to L2.
                     </span>
                   )}
                 </>
               )}
-              <span className="text-xs font-normal text-gray-500">
-                {tokenListInfo}
-              </span>
+              {isL2NativeToken ? (
```

Comment:
instead of a ternary, the two spans' classnames look similar enough that it can be one component